test/java-bdk.spec.js

```diff
@@ -12,153 +13,173 @@ const BASE_PACKAGE = 'com.mycompany.bot'
-                path.join(BASE_JAVA, PACKAGE_DIR, 'OnUserJoinedRoomListener.java'),
-                path.join(BASE_JAVA, PACKAGE_DIR, 'GifSlashHandler.java'),
-                path.join('rsa/privatekey.pem'),
-                path.join('rsa/publickey.pem'),
-                path.join('.gitignore'),
-                path.join(BASE_RESOURCE, 'templates/welcome.ftl'),
-                path.join(BASE_RESOURCE, 'templates/gif.ftl'),
-                path.join(BASE_RESOURCE, 'application.yaml')
-            ]);
-
-        })
+  return helpers.run(path.join(__dirname, '../generators/bdk'))
+    .inTmpDir()
+    .withPrompts({
+      host: 'acme.symphony.com',
+      username: 'test-bot',
+      language: 'Java (beta)',
+      build: 'Gradle',
+      framework: 'spring',
+      groupId: 'com.mycompany',
+      artifactId: 'bot-application',
+      basePackage: BASE_PACKAGE
+    })
+    .then((dir) => {
+      assert.file([
+        'gradlew',
+        'gradlew.bat',
+        'build.gradle',
+        path.join(BASE_JAVA, PACKAGE_DIR, 'BotApplication.java'),
+        path.join(BASE_JAVA, PACKAGE_DIR, 'GifFormActivity.java'),
+        path.join(BASE_JAVA, PACKAGE_DIR, 'OnUserJoinedRoomListener.java'),
+        path.join(BASE_JAVA, PACKAGE_DIR, 'GifSlashHandler.java'),
+        path.join('rsa/privatekey.pem'),
+        path.join('rsa/publickey.pem'),
+        path.join('.gitignore'),
+        path.join(BASE_RESOURCE, 'templates/welcome.ftl'),
+        path.join(BASE_RESOURCE, 'templates/gif.ftl'),
+        path.join(BASE_RESOURCE, 'application.yaml')
+      ]);
+      let privateKey = fs.readFileSync(path.join(dir, 'rsa/privatekey.pem'), 'utf-8')
+      let generatedPublicKey = fs.readFileSync(path.join(dir, 'rsa/publickey.pem'), 'utf-8')
+      assertKeyPair(privateKey, generatedPublicKey)
+    })
 })
 
 it('Generate 2.0 java gradle', () => {
-    return helpers.run(path.join(__dirname, '../generators/bdk'))
-        .inTmpDir()
-        .withPrompts({
-            host: 'acme.symphony.com',
-            username: 'test-bot',
-            language: 'Java (beta)',
-            build: 'Gradle',
-            framework: 'java',
-            groupId: 'com.mycompany',
-            artifactId: 'bot-application',
-            basePackage: BASE_PACKAGE
-        })
-        .then(() => {
-            assert.file([
-                'gradlew',
-                'gradlew.bat',
-                'build.gradle',
-                path.join(BASE_JAVA, PACKAGE_DIR, 'BotApplication.java'),
-                path.join('rsa/privatekey.pem'),
-                path.join('rsa/publickey.pem'),
-                path.join('.gitignore'),
-                path.join(BASE_RESOURCE, 'templates/welcome.ftl'),
-                path.join(BASE_RESOURCE, 'templates/gif.ftl'),
-                path.join(BASE_RESOURCE, 'config.yaml')
-            ]);
-
-        })
+  return helpers.run(path.join(__dirname, '../generators/bdk'))
```

Comment:
we should add an .editorconfig (or a js linter) in this project to avoid formatting differences